app/models/manageiq/providers/embedded_ansible/automation_manager/job.rb

```diff
@@ -111,13 +111,47 @@ def self.collect_authentications(manager, options)
   private_class_method :collect_authentications
 
   def update_with_provider_object(raw_job)
-    self.miq_task ||= raw_job.miq_task
+    transaction do
+      self.miq_task ||= raw_job.miq_task
 
-    update_attributes!(
-      :status      => miq_task.state,
-      :start_time  => miq_task.started_on,
-      :finish_time => raw_status.completed? ? miq_task.updated_on : nil
-    )
+      self.status      = miq_task.state
+      self.start_time  = miq_task.started_on
+      self.finish_time = raw_status.completed? ? miq_task.updated_on : nil
+
+      update_plays
+      save!
+    end
+  end
+
+  def update_plays
+    plays = raw_stdout_json.select do |playbook_event|
+      playbook_event["event"] == "playbook_on_play_start"
+    end.collect do |play|
+      {
+        :name              => play["event_data"]["play"],
+        :resource_status   => play["failed"] ? 'failed' : 'successful',
+        :start_time        => play["created"],
+        :ems_ref           => play["uuid"],
+        :resource_category => "job_play"
+      }
+    end
+
+    # Set each play's finish_time to the next play's start time, with the
+    # final play's finish time set to the entire job's finish time.
+    plays.each_cons(2) do |last_play, play|
+      last_play[:finish_time] = play[:start_time]
+    end
+    plays[-1][:finish_time] = finish_time if plays.any?
+
+    old_resources = resources.index_by(&:ems_ref)
```

Comment:
nice touch to reuse existing resources.

slight concern over not deleting unused resources

Reply:
Yeah, best we can do for now... I really don't want to make changes this close to the build...

😩